### PR TITLE
prefer using dir instead of allinfo for getting smb file info

### PR DESCRIPTION
**Pull request by icewind1991**
some windows server setups don't seem to like the allinfo command use the dir command instead to get the file info where possible.

Also includes some improvements for the timezone detection to properly handle cases where the command errors out.

### CODE REVIEW

File: apps/files_external/3rdparty/icewind/smb/src/TimeZoneProvider.php [hunk 1]
```diff
@@ -35,15 +35,17 @@ public function __construct($host, System $system) {
...
 					$net,
 					escapeshellarg($this->host)
 				);
 				$this->timeZone = exec($command);
-			} else { // fallback to server timezone
-				$this->timeZone = date_default_timezone_get();
 			}
+
+			// fallback to server timezone
+			$this->timeZone = date_default_timezone_get();
```
Comment (rullzer) [inline apps/files_external/3rdparty/icewind/smb/src/TimeZoneProvider.php]: This will just overwrite the stuff above...

Comment (icewind1991) [reply apps/files_external/3rdparty/icewind/smb/src/TimeZoneProvider.php]: Fixed :see_no_evil:

**Review by rullzer — APPROVED** [review]
:elephant:

**Comment by MorrisJobke** [comment]
As this is in 3rdparty: shouldn't upstream be updated first?

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Triggered CI again: https://drone.nextcloud.com/nextcloud/server/9856

**Comment by rullzer** [comment]
@icewind1991 https://drone.nextcloud.com/nextcloud/server/9856/287

**Pull request merged**